velox/functions/prestosql/ArrayPosition.cpp

```diff
@@ -37,24 +37,67 @@ void applyTypedFirstMatch(
   auto rawOffsets = baseArray->rawOffsets();
   auto indices = arrayDecoded.indices();
 
-  rows.applyToSelected([&](auto row) {
-    auto size = rawSizes[indices[row]];
-    auto offset = rawOffsets[indices[row]];
+  if (elementsDecoded.isIdentityMapping() && !elementsDecoded.mayHaveNulls() &&
+      searchDecoded.isConstantMapping()) {
+    if constexpr (std::is_same_v<bool, T>) {
+      auto rawElements = elementsDecoded.values<uint64_t>();
+
+      auto search = bits::isBitSet(searchDecoded.values<uint64_t>(), 0);
+
+      rows.applyToSelected([&](auto row) {
+        auto size = rawSizes[indices[row]];
+        auto offset = rawOffsets[indices[row]];
+
+        int i;
+        for (i = 0; i < size; i++) {
+          if (bits::isBitSet(rawElements, offset + i) == search) {
```

Comment:
here you can even further optimize this if you want, by checking word by word or byte by byte
and check bits only in the corner bytes, if the array is large should save alot. 
i.e: 
if search is 0 then if any byte/word is not 0xFF or 0xFFFFFFF. its found
if search is 1 then if any byte/word is not 0 its found.

but anyway how common is the bool search idk